### PR TITLE
Tweak xray export to improve rendering of traces and improve parity a…

### PR DESCRIPTION
**Pull request by anuraaga**
…gainst native instrumentation.

**Description:** 

Improve parity of xray exporter vs native xray instrumentation

- Don't require a resource to fill in AWS info because instrumentation doesn't always fill it in but it's worth doing a best effort
- Handle AWS attributes as set by java instrumentation
- Remove usage of component which was removed in OTel specification [0.3.0](https://github.com/open-telemetry/opentelemetry-specification/blob/master/CHANGELOG.md#v030-02-21-2020)
- Read net peer IP as client IP
- Set aws namespace for aws operations and span name to service instead of operation
- Don't set namespace to remote for internal spans (spans with parent but not Kind.Client)
- Fix 0 parent ID still set as hex
- Fill SQL info when available rather than filtering on deprecated HTTP component attribute

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** Compared traces in X-Ray console for toy [app](https://github.com/anuraaga/aws-xray-sdk-with-opentelemetry-sample/tree/both-otel-and-xray)

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Waiting for @kbrockhoff to review first :). Definitely they have more experience than I have with x-ray.

**Review by kbrockhoff — APPROVED** [review]
This all looks good to me.

After it gets merged, I will deploy into our development environment and see if any issues arise.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**
